src/infra/src/short_url/mysql.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright 2024 Zinc Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use async_trait::async_trait;
+use sqlx::Row;
+
+use crate::{
+    db::mysql::{create_index, CLIENT},
+    errors::Result,
+    short_url::{ShortUrl, ShortUrlRecord},
+};
+
+pub struct MysqlShortUrl {}
+
+impl MysqlShortUrl {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Default for MysqlShortUrl {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[async_trait]
+impl ShortUrl for MysqlShortUrl {
+    /// Create table short_urls
+    async fn create_table(&self) -> Result<()> {
+        let pool = CLIENT.clone();
+        let query = r#"
+            CREATE TABLE IF NOT EXISTS short_urls (
+                id BIGINT AUTO_INCREMENT PRIMARY KEY,
+                original_url VARCHAR(2048) NOT NULL UNIQUE,
+                short_id VARCHAR(32) NOT NULL UNIQUE,
+                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+            );
```

Comment:
This sql can't create table in my local, MySQL 8.4.2

Reply:
Fixed, made original_url as TEXT and common_idx
`create index original_url_idx on short_urls(original_url(255))`